samples/GalleryApp/Pages/ObjectDetectorPage.xaml.cs

```diff
@@ -400,5 +421,40 @@ private void UIProcessedPreview_SizeChanged(object sender, SizeChangedEventArgs
 
             m_bboxRenderer.ResizeContent(e);
         }
+
+        /// <summary>
+        /// Triggers when all object kind filters are selected
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void SelectAllObjectKinds(object sender, RoutedEventArgs e)
+        {
+            UIObjectKindFilters.SelectAll();
+            FrameSourceStartAsync();
+        }
+
+        /// <summary>
+        /// Triggers when all object kind filters are unselected
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void UnselectAllObjectKinds(object sender, RoutedEventArgs e)
+        {
+            m_objectKinds = null;
+            UIObjectKindFilters.SelectedIndex = -1;
```

Comment:
>UIObjectKindFilters.SelectedIndex = -1; [](start = 12, length = 39)

doesn't this already triggers UIObjectKindFilters_SelectionChanged()? do you still want to trigger the frameReader to start in that case? (contrary to what you do when m_objectKinds.Count == 0 in UIObjectKindFiltersSelectionChanged() )

Reply:
Yes, this implementation will trigger frameReader twice. I changed the logic so that all checkboxes only update once and the frameReader triggers only once when there's an update from TriStateCheckBox or the ObjectKindFilters.

---
In reply to: [308045975](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/49#discussion_r308045975) [](ancestors = 308045975)